endpoints-framework/src/main/java/com/google/api/server/spi/ServiceException.java

```diff
@@ -99,4 +101,17 @@ public String getDomain() {
   public Map<String, String> getHeaders() {
     return null;
   }
+
+  public Level getLogLevel() {
+    return logLevel == null ? getDefaultLoggingLevel(statusCode) : logLevel;
+  }
+
+  private static Level getDefaultLoggingLevel(int statusCode) {
+    return statusCode >= 500 ? Level.SEVERE : Level.INFO;
+  }
+
+  public <T extends ServiceException> T withLogLevel(T exception, Level level) {
```

Comment:
I don't get why this doesn't just modify the log level directly on the instance; How does this even work? `serviceException.withLogLevel(serviceException, Level.INFO)`?

Is there a test that shows usage?

Reply:
This is just syntactic sugar: `throw ServiceException.withLogLevel(new UnauthorizedException(...), Level.INFO);`

Otherwise, you have to do

```UnauthorizedException ex = new UnauthorizedException(...);
ex.setLoggingLevel(Level.SEVERE);
throw ex;```

We can't make a generic mutator without modifying ServiceException to itself be generic or passing in return type class as a parameter.

Reply:
Shouldn't this method be `static`, then?

Reply:
Yes. Added a test.